server/src/main/java/org/opensearch/rest/action/admin/cluster/RestGetDecommissionStateAction.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.opensearch.action.admin.cluster.decommission.awareness.get.GetDecommissionStateRequest;
+import org.opensearch.client.Requests;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.rest.RestRequest.Method.GET;
+
+/**
+ * Returns decommissioned attribute information
+ *
+ * @opensearch.api
+ */
+public class RestGetDecommissionStateAction extends BaseRestHandler {
```

Comment:
`State` and `Status` are used interchangeably. Can we just use one? I prefer Status.

Reply:
We are calling it `_status` in the API as per discussion here in the RFC - #3639. The response that the end user see is nothing but the current status of the decommission action and the attribute which is undergoing decommission. And hence, the status in the API. We are calling the rest class as `RestGetDecommissionStateAction` as the current status and the attribute fetched is directly read from the metadata and given back to the end user and hence the state here.

Also to note, `SnapshotStatus` and `RestSnapshotStatusAction` also follow a similar pattern where in the API consists of `_status` which provides any intermediary states of the snapshot and not just the terminal states

Reply:
+1 on using `status` at all places

Reply:
If I use `Status` here, then it would be inconsistent with the `DELETE` API which is `RestDeleteDecommissionStateAction` which sounds perfect for the delete API. Now for `GET` api we can change it to `Status` but I thought of keeping it in consistent with the DELETE API also `State` here wasn't sounding like misleading as we return the attribute and its status.

Do let me know your preference if you think otherwise